### PR TITLE
VACMS-0000: Fixes some more PHPStan errors.

### PR DESCRIPTION
**Pull request by ndouglas**
## Description

Not done with this yet, but wanted to get tests running quickly since I touched a lot of the test code.  Found an unused (I think) step definition and a try... catch block that I think wasn't actually going to lead to a good outcome, but I might have morning brain 🤷‍♂️ 

**EDIT**: Forget it, I'm done 😃 

## Testing done
This should change very few executed code paths, so the automated tests should be sufficient.

### CODE REVIEW

File: tests/behat/Drupal/ContentModelContext.php [hunk 1]
```diff
@@ -2,16 +2,14 @@
...
-use Behat\Behat\Context\SnippetAcceptingContext;
 use Behat\Gherkin\Node\TableNode;
 use Drupal\DrupalExtension\Context\RawDrupalContext;
 use TravisCarden\BehatTableComparison\TableEqualityAssertion;
 
 /**
  * Provides content model step definitions for Behat.
  */
-class ContentModelContext extends RawDrupalContext implements SnippetAcceptingContext {
+class ContentModelContext extends RawDrupalContext {
```
Comment (ndouglas) [inline tests/behat/Drupal/ContentModelContext.php:12]: SnippetAcceptingContext is deprecated.  It was injected by the generator, and we don't use its features anyway.

File: tests/behat/Drupal/ContentModelContext.php [hunk 2]
```diff
@@ -71,13 +69,13 @@ public function assertAutoLabels(TableNode $expected) {
    *
    * @Then /^I select the "([^"]*)" radio button$/
    *
-   * @throws Behat\Behat\Exception\BehaviorException
+   * @throws \Exception
```
Comment (ndouglas) [inline tests/behat/Drupal/ContentModelContext.php:72]: This exception class does not seem to actually exist.

File: tests/behat/Drupal/ContentModelContextCustom.php [hunk 3]
```diff
@@ -118,14 +118,7 @@ public function assertBundleFields(TableNode $expected, $bundles, $entity_type_i
...
     $fields = [];
-    try {
-      $entity_type = $this->entityTypeManager()->getDefinition($entity_type_id);
-    }
-    catch (PluginNotFoundException $e) {
-      // A PluginNotFoundException here just means that the module providing
-      // the entity type in question isn't installed. Continue.
-    }
-
+    $entity_type = $this->entityTypeManager()->getDefinition($entity_type_id);
```
Comment (ndouglas) [inline tests/behat/Drupal/ContentModelContextCustom.php:121]: I don't think this code made sense as it existed.  If an exception had been thrown setting `$entity_type` and ignored, then another one would surely be called when trying to call a function on `$entity_type` a couple lines later.

Also think "the module providing the entity type in question isn't installed" might be a serious problem in a functional test...

File: tests/behat/Drupal/FeatureContext.php [hunk 4]
```diff
@@ -104,32 +103,6 @@ public function iViewTerm($page, $title) {
...
-    }
-    foreach ($menu_links as $mlid => $menu_link) {
-      $links['menu_name'][] = $menu_link->title->value;
-    }
-
-    if (!in_array($item, $links['menu_name'])) {
-      throw new \Exception('Menu link "' . $item . '" does not exist');
-    }
-  }
-
```
Comment (ndouglas) [inline tests/behat/Drupal/FeatureContext.php:132]: Not called anywhere.

File: tests/behat/Drupal/FeatureContext.php [hunk 5]
```diff
@@ -313,26 +286,28 @@ public function theElementShouldHaveAttributeMatchingPattern($element, $attribut
...
    * @Given I set the node with title :title to :status
    */
   public function setTheNodeWithTitle($status, $title) {
-    if ($status === 'published') {
-      $status = ['pub' => TRUE, 'mod' => 'published'];
-    }
-    else {
-      $status = ['pub' => FALSE, 'mod' => 'archived'];
-    }
-
```
Comment (ndouglas) [inline tests/behat/Drupal/FeatureContext.php:325]: I didn't see much benefit to this.  I just moved the condition later in the function to make the effects immediate and explicit and remove this middleman redefinition.

File: tests/behat/Drupal/FeatureContext.php [hunk 6]
```diff
@@ -452,15 +427,15 @@ public function theOptionFromShouldBeSelected($option, $select) {
...
+   * @throws \Exception
    *
    * @Then :selector should contain :item
    */
   public function thenSelectorShouldContain($selector, $item) {
     $session = $this->getSession();
     $select_item = $session->getPage()->find('css', $selector);
     if (empty($select_item)) {
-      throw new \Exception('The selector ' . $select_item . ' does not exist.');
+      throw new \Exception('The selector ' . $selector . ' does not exist.');
```
Comment (ndouglas) [inline tests/behat/Drupal/FeatureContext.php:438]: if it's empty, printing it isn't going to give us any useful information 😃

File: tests/behat/Drupal/UserRolesAndPermissionsContext.php [hunk 7]
```diff
@@ -1,169 +0,0 @@
-<?php
```
Comment (ndouglas) [inline tests/behat/Drupal/UserRolesAndPermissionsContext.php:1]: Unused.  This was moved to a PHPUnit test.

**Review by indytechcook — APPROVED** [review]

**Pull request merged**